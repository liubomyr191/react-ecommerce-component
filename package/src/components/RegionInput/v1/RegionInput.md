### Overview

The `RegionInput` component renders either a `Select` or `TextInput`, based on the number of region options.

### Usage

#### Input in a form

```jsx
const locales = {
  "NR": {
    "name": "Country Without Regions",
  },
  "WR": {
    "name": "Country With Regions",
    "states": {
      "AA": {
        "name": "A State"
      },
      "BB": {
        "name": "B State"
      }
    }
  }
};

<AddressForm isOnDarkBackground locales={locales} />
```

#### Input with a value

```jsx
<RegionInput isOnDarkBackground isReadOnly value="California" />
```

### Theme

Assume that any theme prop that does not begin with "rui" is within `rui_components`. See [Theming Components](./#!/Theming%20Components).

| Theme Prop | Default | Description |
