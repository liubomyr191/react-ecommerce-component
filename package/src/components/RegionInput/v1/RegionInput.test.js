import React from "react";
import renderer from "react-test-renderer";
import mockComponents from "../../../tests/mockComponents";
import RegionInput from "./RegionInput";

test("basic snapshot with only the components and required props should render a TextInput", () => {
  const component = renderer.create(<RegionInput name="region" components={mockComponents} />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("basic snapshot with other form props", () => {
  const component = renderer.create(<RegionInput components={mockComponents} value="California" name="region" isReadOnly />);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("basic snapshot with pre-filled value in TextInput", () => {
  const component = renderer.create(<RegionInput name="region" components={mockComponents} value="California"/>);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("basic snapshot with pre-filled value in Select", () => {
  const options = [
    {
      value: "AA",
      label: "A State"
    },
    {
      value: "BB",
      label: "B State"
    }
  ];

  const component = renderer.create(<RegionInput name="region" components={mockComponents} options={options} value="BB"/>);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("basic snapshot with options should render a Select", () => {
  const options = [
    {
      value: "AA",
      label: "A State"
    },
    {
      value: "BB",
      label: "B State"
    }
  ];

  const component = renderer.create(<RegionInput name="region" components={mockComponents} options={options}/>);

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
