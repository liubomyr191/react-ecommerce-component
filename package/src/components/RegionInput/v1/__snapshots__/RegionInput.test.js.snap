// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot with only the components and required props should render a TextInput 1`] = `"TextInput(undefined)"`;

exports[`basic snapshot with options should render a Select 1`] = `"Select(undefined)"`;

exports[`basic snapshot with other form props 1`] = `"TextInput(undefined)"`;

exports[`basic snapshot with pre-filled value in Select 1`] = `"Select(undefined)"`;

exports[`basic snapshot with pre-filled value in TextInput 1`] = `"TextInput(undefined)"`;
